rocAL/rocAL/source/meta_data/bounding_box_graph.cpp

```diff
@@ -232,14 +232,81 @@ void BoundingBoxGraph::update_box_encoder_meta_data(std::vector<float> *anchors,
                 else
                 {
                     //Convert the "ltrb" format to "xcycwh"
-                    encoded_bb[anchor_idx].xcycwh.xc = 0.5 * (p_anchor->ltrb.l + p_anchor->ltrb.r); //xc
-                    encoded_bb[anchor_idx].xcycwh.yc = 0.5 * (p_anchor->ltrb.t + p_anchor->ltrb.b); //yc
-                    encoded_bb[anchor_idx].xcycwh.w = (-p_anchor->ltrb.l + p_anchor->ltrb.r);      //w
-                    encoded_bb[anchor_idx].xcycwh.h = (-p_anchor->ltrb.t + p_anchor->ltrb.b);      //h
+                    encoded_bb[anchor_idx].xc = 0.5 * (p_anchor->l + p_anchor->r); //xc
+                    encoded_bb[anchor_idx].yc = 0.5 * (p_anchor->t + p_anchor->b); //yc
+                    encoded_bb[anchor_idx].w = (-p_anchor->l + p_anchor->r);      //w
+                    encoded_bb[anchor_idx].h = (-p_anchor->t + p_anchor->b);      //h
                     encoded_labels[anchor_idx] = 0;
                 }
             }
         }
     }
 }
 
+void BoundingBoxGraph::update_box_iou_matcher(
+    std::vector<float> *anchors, int *matches_idx_buffer,
+    pMetaDataBatch full_batch_meta_data, float criteria, float high_threshold,
+    float low_threshold, bool allow_low_quality_matches) 
+{
+    auto bb_coords_batch = full_batch_meta_data->get_bb_cords_batch();
+    unsigned anchors_size = anchors->size() / 4;  // divide the anchors_size by 4 to get the total number of anchors
+    BoundingBoxCord *bbox_anchors = reinterpret_cast<BoundingBoxCord *>(anchors->data());
+
+    std::vector<int *> matches(full_batch_meta_data->size());
+    for (int i = 0; i < full_batch_meta_data->size(); i++) {
+        matches[i] = reinterpret_cast<int *>(matches_idx_buffer + i * anchors_size);
+    }
+
+#pragma omp parallel for
+    for (int i = 0; i < full_batch_meta_data->size(); i++) {
+        auto bb_coords = bb_coords_batch[i];
+        auto bb_count = bb_coords.size();
+
+        std::vector<float> matched_vals(anchors_size, -1.0);
+        std::vector<int> low_quality_preds(anchors_size, -1);
+
+        // Calculate IoU's, The number of IoU Values calculated will be (bb_count x anchors_size)
+        for (unsigned bb_idx = 0; bb_idx < bb_count; bb_idx++) {
+            BoundingBoxCord box = bb_coords[bb_idx];
+            float box_area = (box.b - box.t) * (box.r - box.l);
+            float best_bbox_iou = -1.0f;
+            std::vector<float> bbox_iou(anchors_size);  // IoU value for bbox mapped with each anchor
+            for (unsigned int anchor_idx = 0; anchor_idx < anchors_size; anchor_idx++) {
+                float iou_val = ssd_BBoxIntersectionOverUnion(box, box_area, bbox_anchors[anchor_idx]);
```

Comment:
This naming of the functiom seems to contain both snake and camel case.
Change to snake_case

Reply:
done